mkt/search/tests/test_views.py

```diff
@@ -608,22 +614,48 @@ def test_no_flash_on_firefoxos(self):
         f = self.webapp.get_latest_file()
         f.uses_flash = True
         f.save()
-        self.webapp.save()
-        self.refresh('webapp')
+        self.reindex(Webapp)
         res = self.anon.get(self.url, data={'dev': 'firefoxos'})
         eq_(res.status_code, 200)
         eq_(len(res.json['objects']), 0)
 
     def test_premium_types(self):
-        res = self.anon.get(self.url,
-                            data={'premium_types': 'free'})
+        with self.assertNumQueries(0):
+            res = self.anon.get(self.url, data={'premium_types': 'free'})
+        eq_(res.status_code, 200)
+        obj = res.json['objects'][0]
+        eq_(obj['slug'], self.webapp.app_slug)
+
+    def test_premium_type_premium(self):
+        # Make the app paid.
+        self.make_premium(self.webapp)
+        self.user = UserProfile.objects.get(pk=2519)
+
+        # Payment account stuff. We need to create one to avoid useless queries
+        # made when we can't find the payment account in Webapp model.
+        self.seller = SolitudeSeller.objects.create(
+            resource_uri='/path/to/sel', uuid='seller-id', user=self.user)
+        self.account = PaymentAccount.objects.create(
+            user=self.user, uri='asdf', name='test', inactive=False,
+            solitude_seller=self.seller, account_id=123)
+        AddonPaymentAccount.objects.create(
+            addon=self.webapp, account_uri='foo',
+            payment_account=self.account, product_uri='bpruri')
+
+        # Reindex once we have everything.
+        self.reindex(Webapp)
+
+        # There should (sadly) be 2 queries: one for the AddonPremium model and
+        # price, and one for the AddonPaymentAccount.
+        with self.assertNumQueries(2):
+            res = self.anon.get(self.url, data={'premium_types': 'premium'})
         eq_(res.status_code, 200)
         obj = res.json['objects'][0]
         eq_(obj['slug'], self.webapp.app_slug)
```

Comment:
We already have lots of tests in `mkt/webapps/tests/test_serializers.py` that make sure the serializer output has not changed so I didn't think we'd need to do more than that.